src/benchmarks/micro/libraries/Microsoft.Extensions.Configuration/GetChildKeysTests.cs

```diff
@@ -0,0 +1,111 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Configuration.Memory;
+using BenchmarkDotNet.Attributes;
+
+namespace Benchmark
+{
+    [BenchmarkCategory(Categories.Libraries)]
+    public class GetChildKeysTests
+    {
+        private ChainedConfigurationProvider _chainedConfig;
+        private ChainedConfigurationProvider _chainedConfigEmpty; 
+        private ChainedConfigurationProvider _chainedConfigWithSplitting;
+        private ChainedConfigurationProvider _chainedConfigWithCommonPaths;
+
+        [GlobalSetup(Targets = new[] {
+            nameof(AddChainedConfigurationNoDelimiter),
+            nameof(AddChainedConfigurationEmpty),
+            nameof(AddChainedConfigurationWithSplitting),
+            nameof(AddChainedConfigurationWithCommonPaths)
+        })]
+        public void SetupBasic()
+        {
+            var emptyKeys = new Dictionary<string, string>() { };
+            for (int i = 0; i < 1000; i++)
+            {
+                emptyKeys.Add(new string(' ', i), string.Empty);
+            }
+
+            var inputKeys = new Dictionary<string, string>() { };
+            for (int i = 1000; i < 2000; i++)
+            {
+                inputKeys.Add(i.ToString(), string.Empty);
+            }
+
+            var splittingKeys = new Dictionary<string, string>() { };
+            for (int i = 1000; i < 2000; i++)
+            {
+                splittingKeys.Add("a:" + i.ToString(), string.Empty);
+            }
+
+            var keysWithCommonPaths = new Dictionary<string, string>() { };
+            for (int i = 1000; i < 2000; i++)
+            {
```

Comment:
I wonder whether we could use more realistic input (I doubt that our users have 1k of very short keys in the config).

@davidfowl where could we find some real life examples of user configs?